### ISSUE TITLE
[Neat URL] Open link in new tab problems (addons.mozilla.org #42)

### ISSUE THREAD

**Issue by gwarser** [Smile4ever/firefoxaddons#48]
As you requested in https://github.com/Smile4ever/firefoxaddons/issues/42

Firefox stable, clean profile, only last version of Neat URL installed.

Open browser console.

1. Open this link in new tab https://blog.mozilla.org/firefox/introducing-firefox-multi-account-containers/
2. Scroll to first link in article "The Firefox Multi-Account Containers extension" `https://addons.mozilla.org/firefox/addon/multi-account-containers/?utm_source=blog.mozilla.org&utm_campaign=firefox_frontier&utm_medium=referral`
3. Open this link in new tab (middle click or from context menu)

New tab will open, and 
- current tab will be redirected to above link, but without utm parameters, new tab will have URL with utm parameters
**OR**
- if you go to this new tab, you will see additional entry in history under back button containing URL with utm parameters

Browser console will show this error:

```
[Exception... "Illegal value"  nsresult: "0x80070057 (NS_ERROR_ILLEGAL_VALUE)"  location: "JS frame :: chrome://browser/content/ext-tabs.js :: query :: line 487"  data: no]  (nieznane)
	query chrome://browser/content/ext-tabs.js:487:29
	next self-hosted:1183:9
	query self-hosted:947:17
	call/result< resource://gre/modules/ExtensionParent.jsm:696:57
	withPendingBrowser resource://gre/modules/ExtensionParent.jsm:395:26
	InterpretGeneratorResume self-hosted:1276:8
	next self-hosted:1183:9
	call resource://gre/modules/ExtensionParent.jsm:695:20
	InterpretGeneratorResume self-hosted:1276:8
	next self-hosted:1183:9
Error: An unexpected error occurred  undefined
```
In Nightly this is:
```
[Exception... "Illegal value"  nsresult: "0x80070057 (NS_ERROR_ILLEGAL_VALUE)"  location: "JS frame :: chrome://browser/content/ext-tabs.js :: query :: line 521"  data: no]  (unknown)
	query chrome://browser/content/ext-tabs.js:521:29
	InterpretGeneratorResume self-hosted:1283:8
	next self-hosted:1190:9
```
Screencast:
https://gfycat.com/PinkWeepyAmericanrobin

**Comment by Smile4ever** [comment]
Thank you, I have been able to reproduce. I will investigate and report back.

**Comment by Smile4ever** [comment]
This has been fixed in Neat URL 3.1.0. Thank you.

**Comment by gwarser** [comment]
Sorry, I cannot confirm. Tested just now 3.1.0 on clean ESR, stable and Nightly and I see no difference :(

**Comment by Smile4ever** [comment]
Reopening this to investigate further.

**Comment by Smile4ever** [comment]
Please try Neat URL 3.1.1.

**Comment by gwarser** [comment]
I no longer see `[Exception... "Illegal value"...` in console.

Still, both tabs are redirected to new link.

I just discovered that both tabs are redirected to localized content.
Original URL have `...mozilla.org/firefox...`
In new tabs this is `...mozilla.org/pl/firefox...` or `...mozilla.org/en-US/firefox...` (depends on browser locale)

Browser console shows two "cleaning tasks", one original, and one localized:

Stable, pl:
```
"Neat URL (type main_frame): https://addons.mozilla.org/firefox/addon/multi-account-containers/?utm_source=blog.mozilla.org&utm_campaign=firefox_frontier&utm_medium=referral has been changed to https://addons.mozilla.org/firefox/addon/multi-account-containers/"  background.js:546:3
"Neat URL (type main_frame): https://addons.mozilla.org/pl/firefox/addon/multi-account-containers/?utm_source=blog.mozilla.org&utm_campaign=firefox_frontier&utm_medium=referral has been changed to https://addons.mozilla.org/pl/firefox/addon/multi-account-containers/"  background.js:546:3
```

ESR, en_US:
```
"Neat URL (type main_frame): https://addons.mozilla.org/firefox/addon/multi-account-containers/?utm_source=blog.mozilla.org&utm_campaign=firefox_frontier&utm_medium=referral has been changed to https://addons.mozilla.org/firefox/addon/multi-account-containers/"  background.js:546:3
"Neat URL (type main_frame): https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/?utm_source=blog.mozilla.org&utm_campaign=firefox_frontier&utm_medium=referral has been changed to https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/"  background.js:546:3
```

**Comment by gwarser** [comment]
There is settings in about:config named `privacy.resistFingerprinting.block_mozAddonManager`. When set to true, extensions should be allowed on addons.mozilla.org. 
Even when it's set to true, I still experience this bug.

**Comment by Smile4ever** [comment]
Neat URL tries to support addons.mozilla.org without flipping that prefby doing something different. That's the reason when the pref is flipped, there is stil a bug with addons.mozilla.org in 3.1.0. This has been fixed in Neat URL 3.2.0.

**Comment by gwarser** [comment]
It's fixed. Thank you.
